### PR TITLE
SDL2_image_2_0: use finalAttrs, remove version,hash from inputs, remove patch version from attr

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -9258,7 +9258,10 @@ self: super: with self; {
 
   piccata = callPackage ../development/python-modules/piccata { };
 
-  pick = callPackage ../development/python-modules/pick { };
+  pygame = callPackage ../development/python-modules/pygame {
+    inherit (pkgs.darwin.apple_sdk.frameworks) AppKit;
+    SDL2_image = pkgs.SDL2_image_2_0_5;
+  };
```
Comment (NickCao) [inline pkgs/top-level/python-packages.nix:9264]: Please fix this part.

Comment (NickCao) [reply pkgs/top-level/python-packages.nix:9264]: This wrongfully replaces the `pick` attribute.

Comment (SuperSandro2000) [reply pkgs/top-level/python-packages.nix:9264]: Bad rebased on my end

**Review by NickCao — APPROVED** [review]
This removes a top level attribute `SDL2_image_2_0_5` which normally guarantees an entry in aliases.nix, but since it's most likely only used in nixpkgs, I'd give it a pass.

Result of `nixpkgs-review pr 217164` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>SDL2_image_2_0</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
IIRC we don't do entries in aliases.nix for versioned attributes like this.

**Pull request merged**